File: mqtt-old/src/main/java/org/sheedon/mqtt/Call.java
package org.sheedon.mqtt;

/**
 * 调度封装的基本接口
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2020/2/11 12:46
 */
public interface Call extends Cloneable {
    /**
     * Returns the original request that initiated this call.
     */
    Request request();


    void publishNotCallback();


    <R extends Response> void enqueue(Callback<R> callback);

    /**
     * Cancels the request, if possible. Requests that are already complete cannot be canceled.
     */
    void cancel();


    boolean isExecuted();

    boolean isCanceled();


    Call clone();
}

File: mqtt/src/main/java/org/sheedon/mqtt/Call.kt
package org.sheedon.mqtt

import org.sheedon.rr.core.Call

/**
 * A call is a request that is ready to execute.
 * Since the object represents a single request-response pair (stream), it cannot be executed twice.
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/1/30 7:43 下午
 */
interface Call : Call<String, RequestBody, ResponseBody> {

    /**
     * Use the Callback in the protocol, you can also use the default enqueue (RRCallback callback),
     * but too many generics are displayed
     *
     * @param callback Callback with request
     */
    fun enqueue(callback: Callback?)

}